src/editors/containers/VideoEditor/components/VideoSettingsModal/components/DurationWidget/hooks.js

```diff
@@ -0,0 +1,219 @@
+import { useEffect, useState } from 'react';
+import { useSelector } from 'react-redux';
+
+import { actions, selectors } from '../../../../../../data/redux';
+import messages from '../messages';
+
+import * as module from './hooks';
+
+const durationMatcher = /^(\d{0,2}):?(\d{0,2})?:?(\d{0,2})?$/i;
+
+export const durationWidget = ({ dispatch }) => {
```

Comment:
The name is a bit ambiguous because now you have a component and a function with the same name. Also, this function calls other hooks inside it (like `useSelector`), which makes it a custom hook. That is why the name must start with `use`. (For example, `useDuration` would probably be a good name). I know we don't name custom hooks that way so far in the project, but that is a mistake - have a look at the React docs here (https://reactjs.org/docs/hooks-custom.html) if you want to know more. Since it is important to distinguish between custom hooks and other functions, React makes it very clear that it is mandatory to name custom hooks with `use`. If we don't do that, it will be confusing and make other React developers have problems understanding our code.

Reply:
I've changed the code to use connect instead of useSelector now.